### PR TITLE
Feature/implement routine flow reports

### PR DESCRIPTION
**Pull request by miatrinity**

### CODE REVIEW

File: app/controllers/routine_flow_reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# :nodoc:
+class RoutineFlowReportsController < ApplicationController
+  def index
+    load_routine_flows
+  end
+
+  def show
+    load_routine_flow
+    create_routine_flow_report_presenter
```
Comment (MiaSinek) [inline app/controllers/routine_flow_reports_controller.rb]: Refactor this with view-based presenter

File: app/controllers/routine_flow_reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  private
+
+  def load_routine_flows
+    @routine_flows ||= routine_flow_scope.order('completed_at desc').to_a
```
Comment (MiaSinek) [inline app/controllers/routine_flow_reports_controller.rb]: This is not standard - how can we make this better?

File: app/models/routine_flow.rb [hunk 3]
```diff
@@ -12,6 +14,7 @@ class RoutineFlow < ApplicationRecord
   def start!
     clone_steps_of routine
     activate_first_flow_step
+    update_start_time
   end
```
Comment (MiaSinek) [inline app/models/routine_flow.rb]: Inconsistent usage of bangs - `take_next_flow_step!` and `complete_routine_flow!`  are calling bang methods, this one doesn't.

Make it consistent!

File: app/models/routine_flow.rb [hunk 4]
```diff
@@ -12,6 +14,7 @@ class RoutineFlow < ApplicationRecord
   def start!
```
Comment (MiaSinek) [inline app/models/routine_flow.rb]: Refactor - move this all out. Thoughts: https://3.basecamp.com/4370054/buckets/15565958/todos/2578977166

File: app/views/routine_flows/_routine_flow.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<%# 🏆 #3 | 📅 #{routine_flow1.completed_at.to_s(:long_ordinal)} | ⏱️ 2h 17m 25s %>
```
Comment (MiaSinek) [inline app/views/routine_flows/_routine_flow.html.erb]: Remove this comment

File: spec/models/routine_flow_spec.rb [hunk 6]
```diff
@@ -40,15 +43,43 @@
...
+      start_time = Time.current - 7.minutes
+      finish_time = start_time + 7.minutes
+
+      travel_to start_time
+
+      routine, routine_flow = start_routine_flow
+
+      travel_to finish_time
+
+      (routine.steps.count-1).times { routine_flow.take_next_flow_step! }
```
Comment (MiaSinek) [inline spec/models/routine_flow_spec.rb]: Swap these 2 lines - easier to understand test.

**Review by MiaSinek — CHANGES_REQUESTED** [review]

File: app/controllers/routine_flow_reports_controller.rb (hunk 2 above)
Comment (miatrinity) [reply app/controllers/routine_flow_reports_controller.rb]: Moved it to routine_flow_scope

File: app/models/routine_flow.rb (hunk 3 above)
Comment (miatrinity) [reply app/models/routine_flow.rb]: Removed all unnecessary bangs in our code!

File: app/views/routine_flows/_routine_flow.html.erb (hunk 5 above)
Comment (miatrinity) [reply app/views/routine_flows/_routine_flow.html.erb]: Done

File: spec/models/routine_flow_spec.rb (hunk 6 above)
Comment (miatrinity) [reply spec/models/routine_flow_spec.rb]: Done, boss!

File: spec/business_processes/routine_flow_stepper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    #   expect(routine_flow.started_at).to eq(Time.current)
+    # end
+
+
+    # it 'activates first flow step' do
+    #   _, routine_flow = start_routine_flow
+
+    #   expect(routine_flow.flow_steps.first.status).to eql('active')
+    # end
```
Comment (MiaSinek) [inline spec/business_processes/routine_flow_stepper_spec.rb]: Uncomment this

**Review by MiaSinek — CHANGES_REQUESTED** [review]

File: spec/business_processes/routine_flow_stepper_spec.rb (hunk 7 above)
Comment (miatrinity) [reply spec/business_processes/routine_flow_stepper_spec.rb]: OK

**Review by MiaSinek — APPROVED** [review]

**Pull request merged**